sql/transactions.go

```diff
@@ -0,0 +1,177 @@
+package sql
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/spacemeshos/sqlite"
+)
+
+// TxRetryOpt for configuring db transactions.
+type TxRetryOpt func(tr *txRetryer)
+
+type txRetryer struct {
+	retryNum int
+	retryMs  time.Duration
+}
+
+func newTxRetryer(opts ...TxRetryOpt) *txRetryer {
+	tr := &txRetryer{
+		retryMs:  10,
+		retryNum: 3,
+	}
+	for _, opt := range opts {
+		opt(tr)
+	}
+	return tr
+}
+
+// Tx is wrapper for database transaction.
+type Tx struct {
+	db        *Database
+	conn      *sqlite.Conn
+	committed bool
+	err       error
+}
+
+// WithRetryBusy number of retry db transactions.
+func WithRetryBusy(n int) TxRetryOpt {
+	return func(t *txRetryer) {
+		t.retryNum = n
+	}
+}
+
+// WithRetryTimeout timeout for retrying db transactions.
+func WithRetryTimeout(milliseconds int) TxRetryOpt {
```

Comment:
i think it is always better to use time.Duration for anything related to duration in golang